test/models/mentor/discussion_test.rb

```diff
@@ -268,4 +270,81 @@ class Mentor::DiscussionTest < ActiveSupport::TestCase
     assert_equal "User in Anonymous mode", discussion.student_name
     assert_nil discussion.student_avatar_url
   end
+
+  test "recalculates num_solutions_mentored" do
+    mentor = create :user
+
+    perform_enqueued_jobs do
+      discussion_1 = create :mentor_discussion, mentor: mentor
+      discussion_2 = create :mentor_discussion, mentor: mentor
+      discussion_3 = create :mentor_discussion, mentor: mentor
+
+      # Sanity check
+      assert_equal 0, mentor.num_solutions_mentored
+
+      discussion_1.student_finished!
+      assert_equal 1, mentor.reload.num_solutions_mentored
+
+      discussion_2.finished!
+      assert_equal 2, mentor.reload.num_solutions_mentored
+
+      discussion_3.finished!
+      assert_equal 3, mentor.reload.num_solutions_mentored
+    end
```

Comment:
I've kep this test as an integration test that things are continually being updated

Reply:
Good decision. I think it would also make sense to add tests for the commands too (as I can see them being used outside of this scenario). They could just be copied/pasted from here. Not blocking merge though.